src/tbb/dynamic_link.cpp

```diff
@@ -328,7 +326,11 @@ namespace r1 {
 #endif  // __TBB_DYNAMIC_LOAD_ENABLED
     void init_dynamic_link_data() {
     #if __TBB_DYNAMIC_LOAD_ENABLED
-        std::call_once( init_dl_data_state, init_dl_data );
+        static bool init = []() -> bool {
+            init_dl_data();
+            return true;
+        }();
+        (void)init;
```

Comment:
```suggestion
        static bool init = [] {
            init_dl_data();
            return true;
        }();
       suppress_unused_warning(init);
```
Could you also add comment why this approach is used?